### PR TITLE
Remove unauthenticated behavior

### PR DESCRIPTION
**Pull request by junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Remove unauthenticated behavior

## Changelog

- Improvement

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/5257

### CODE REVIEW

File: query/src/servers/http/middleware.rs [hunk 1]
```diff
@@ -38,14 +39,22 @@ pub struct HTTPSessionMiddleware {
...
         let msg = &format!("Multiple {} headers detected", AUTHORIZATION);
         return Err(ErrorCode::AuthenticateFailure(msg));
     }
     if auth_headers.is_empty() {
-        return Ok(None);
+        return Err(ErrorCode::AuthenticateFailure(
+            "No authorization header detected",
+        ));
+    }
+    let mut hostname = headers.get(HOST).unwrap().to_str().unwrap_or("%");
```
Comment (BohuTANG) [inline query/src/servers/http/middleware.rs:53]: Do we need to handle the unwrap ?

File: query/tests/it/servers/http/http_query_handlers.rs [hunk 2]
```diff
@@ -424,15 +424,15 @@ async fn test_query_log() -> Result<()> {
...
-    assert!(
-        result.data[0][3]
-            .as_str()
-            .unwrap()
-            .to_lowercase()
-            .contains("backtrace"),
-        "{:?}",
-        result
-    );
+    // assert!(
```
Comment (BohuTANG) [inline query/tests/it/servers/http/http_query_handlers.rs:427]: stale codes or temporary fix?

**Comment by youngsofun** [comment]
I think we should use https://docs.rs/poem/latest/poem/web/struct.RemoteAddr.html

File: query/tests/it/servers/http/http_query_handlers.rs [hunk 3]
```diff
@@ -543,13 +546,16 @@ async fn post_json_to_endpoint(
...
     let body = serde_json::to_vec(&json)?;
+    let basic = headers::Authorization::basic("root", "");
 
     let response = ep
         .call(
             Request::builder()
                 .uri(uri.parse().unwrap())
                 .method(Method::POST)
                 .header(header::CONTENT_TYPE, content_type)
+                .header(header::HOST, "localhost")
```
Comment (youngsofun) [inline query/tests/it/servers/http/http_query_handlers.rs:557]: not safe  if client can set it

File: query/tests/it/servers/http/http_query_handlers.rs (hunk 2 above)
Comment (junnplus) [reply query/tests/it/servers/http/http_query_handlers.rs]: Sorry, this is debugging locally.

File: query/tests/it/servers/http/http_query_handlers.rs (hunk 3 above)
Comment (junnplus) [reply query/tests/it/servers/http/http_query_handlers.rs]: I understand that the client cannot set it, but I removed it.

File: query/src/servers/http/middleware.rs (hunk 1 above)
Comment (junnplus) [reply query/src/servers/http/middleware.rs]: I changed to use `remote_addr` of request.

File: query/src/servers/http/middleware.rs [hunk 4]
```diff
@@ -57,21 +64,21 @@ fn get_credential(headers: &HeaderMap) -> Result<Option<Credential>> {
                 let c = Credential::Password {
                     name,
                     password,
-                    hostname: None,
+                    hostname: client_ip,
```
Comment (youngsofun) [inline query/src/servers/http/middleware.rs:67]: can u1@<some_ip> fallback  to  "u1"@"%"` too?

Comment (youngsofun) [reply query/src/servers/http/middleware.rs:67]: do we always use the IP/hostname as a part of  user_id?

or

just may localhost special? and let other addrs map to "%" directly?

Comment (junnplus) [reply query/src/servers/http/middleware.rs:67]: > can u1@<some_ip> fallback to "u1"@"%"` too?

Yes. see https://github.com/datafuselabs/databend/blob/main/query/src/users/user_mgr.rs#L69-L75

File: query/src/servers/http/middleware.rs [hunk 5]
```diff
@@ -38,15 +38,22 @@ pub struct HTTPSessionMiddleware {
...
     }
     if auth_headers.is_empty() {
-        return Ok(None);
+        return Err(ErrorCode::AuthenticateFailure(
+            "No authorization header detected",
+        ));
     }
+    let client_ip = match req.remote_addr().0 {
+        Addr::SocketAddr(addr) => Some(addr.ip().to_string()),
+        Addr::Custom(..) => Some("127.0.0.1".to_string()),
```
Comment (youngsofun) [inline query/src/servers/http/middleware.rs:54]: where is this case used

**Review by youngsofun — APPROVED** [review]

File: query/src/servers/http/middleware.rs (hunk 5 above)
Comment (junnplus) [reply query/src/servers/http/middleware.rs:54]: For testing,  `ep.call` will not have remote addr

**Pull request merged**